### PR TITLE
Add an universal endpoint for images comparison

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Based on https://github.com/appium/appium-support/pull/65

### CODE REVIEW

File: lib/basedriver/commands/images.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const commands = {}, helpers = {}, extensions = {};
+
+const MATCH_FEATURES_MODE = 'matchFeatures';
+const GET_SIMILARITY_MODE = 'getSimilarity';
+const MATCH_TEMPLATE_MODE = 'matchTemplate';
+
+commands.compareImages = async function (mode, firstImageData, secondImageData, options = {}) {
+  const img1 = Buffer.from(firstImageData, 'base64');
+  const img2 = Buffer.from(secondImageData, 'base64');
+  switch (_.toLower(mode)) {
```
Comment (jlipps) [inline lib/basedriver/commands/images.js]: why use both `_.toLower` and `String.toLowerCase`?

File: lib/protocol/routes.js [hunk 2]
```diff
@@ -613,6 +613,12 @@ const METHOD_MAP = {
       ]},
     }
   },
+  '/wd/hub/session/:sessionId/appium/compare_images': {
+    POST: {command: 'compareImages', payloadParams: {
+      required: ['mode', 'firstImageData', 'secondImageData'],
```
Comment (jlipps) [inline lib/protocol/routes.js]: thinking about the param names, why not just `firstImage` and `secondImage`, or `image1` and `image2`? We'll specify the data type as b64-encoded PNG, but the fact that it is data seems superfluous.

File: lib/basedriver/commands/images.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import _ from 'lodash';
+import { errors } from '../../protocol/errors';
+import { imageUtil } from 'appium-support';
```
Comment (jlipps) [inline lib/basedriver/commands/images.js]: do we need to update package.json to ensure `appium-support` is at a version which has this library method?

Comment (mykola-mokhnach) [reply lib/basedriver/commands/images.js]: yep, I'm waiting until the module is published

**Pull request merged**